openlibrary/plugins/worksearch/code.py

```diff
@@ -167,6 +182,40 @@ def parse_query_fields(q):
             isbn = normalize_isbn(v)
             if isbn:
                 v = isbn
+        if field_name == 'lcc':
+            # e.g. lcc:[NC1 TO NC1000] to lcc:[NC-0001.00000000 TO NC-1000.00000000]
+            # for proper range search
+            m = re_range.match(v)
```

Comment:
suggestion: move body of `lcc` and `ddc` conditionals into functions (no change / refactor of body behavior). The only goal is to help reduce cognitive load of trying to determine which variables are relevant, modified, in scope, etc.

Reply:
Fixed